### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Ihmoda**

### CODE REVIEW

File: curriculum/js2/8.js [hunk 1]
```diff
@@ -4,9 +4,16 @@
  */
 
 const solution = () => {
-  Array.prototype.gsMap = function() {
-    // To get the actual array, use this
-    return [];
+  Array.prototype.gsMap = function(callback) {
+    return (function gsMap(arr, callback, idx, result) {
```
Comment (songz) [inline curriculum/js2/8.js]: nice hack, but no. 
```
Array.prototype.gsMap = function(cb, result=[]) {
  if(result.length === this.length) return result;
  result.push( cb(this[result.length]) );
}
```

Comment (songz) [reply curriculum/js2/8.js]: try to figure out why.

File: curriculum/js2/8.js [hunk 2]
```diff
@@ -4,9 +4,10 @@
  */
 
 const solution = () => {
-  Array.prototype.gsMap = function() {
-    // To get the actual array, use this
-    return [];
+  Array.prototype.gsMap = function(cb, result = []) {
```
Comment (seemcat) [inline curriculum/js2/8.js:7]: which algorithm problem is this for?  also, we don't allow built in methods to be used.

**Review by seemcat — CHANGES_REQUESTED** [review]

File: curriculum/js2/8.js (hunk 2 above)
Comment (Ihmoda) [reply curriculum/js2/8.js:7]: curriculum/js2/8.js

Comment (Ihmoda) [reply curriculum/js2/8.js:7]: Replicate Array.prototype.map function and call it gsMap

**Comment by Ihmoda** [comment]
Original Problem: 

/**
 * Replicate Array.prototype.map function and call it gsMap
 * @returns {[]}
 */

const solution = () => {
  Array.prototype.gsMap = function () {
    // To get the actual array, use this
    return [];
  };
};

module.exports = {
  solution,

**Review by songz — APPROVED** [review]

**Pull request merged**